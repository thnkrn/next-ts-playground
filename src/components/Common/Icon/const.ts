const prefix = '/icons/icons-sprite.svg#'

export const ICONS = {
  location: `${prefix}location`,
}
