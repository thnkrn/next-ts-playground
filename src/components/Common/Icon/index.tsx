import { styled } from '@compiled/react'
import { useEffect, useState } from 'react'

interface Props {
  src: string
}

const StyledIconWrapper = styled.svg`
  width: 24px;
  height: 24px;
  opacity: 0;

  &.loaded {
    opacity: 1;
  }
`

const Icon = ({ src }: Props) => {
  const [isLoading, setIsLoading] = useState(true)

  useEffect(() => {
    setIsLoading(false)
  }, [])

  return (
    <StyledIconWrapper className={`icon${isLoading ? '' : ' loaded'}`}>
      <use href={src} />
    </StyledIconWrapper>
  )
}

export default Icon
