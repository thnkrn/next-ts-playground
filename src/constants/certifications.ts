export const CERTIFICATIONS_LIST = [
  {
    issuing: 'HackerRank',
    cerList: [
      {
        name: 'React Certificate',
        issuedDate: 'Oct 2020',
        img: 'reb',
        link: 'https://www.hackerrank.com/certificates/a297b77ebaae',
      },
      {
        name: 'JavaScript (Basic) Certificate',
        issuedDate: 'Nov 2021',
        img: 'jsb',
        link: 'https://www.hackerrank.com/certificates/291200fc4aeb',
      },
      {
        name: 'JavaScript (Intermediate) Certificate',
        issuedDate: 'Nov 2021',
        img: 'jsi',
        link: 'https://www.hackerrank.com/certificates/547359a09d6b',
      },
      {
        name: 'Problem Solving (Basic) Certificate',
        issuedDate: 'Oct 2020',
        img: 'psb',
        link: 'https://www.hackerrank.com/certificates/f9a857ead1cd',
      },
      {
        name: 'Problem Solving (Intermediate) Certificate',
        issuedDate: 'Nov 2021',
        img: 'psi',
        link: 'https://www.hackerrank.com/certificates/58d6c2995949',
      },
      {
        name: 'CSS Certificate',
        issuedDate: 'Nov 2021',
        img: 'css',
        link: 'https://www.hackerrank.com/certificates/b2178802e345',
      },
      {
        name: 'SQL (Basic) Certificate',
        issuedDate: 'Nov 2021',
        img: 'sqlb',
        link: 'https://www.hackerrank.com/certificates/11f8cf438ea4',
      },
      {
        name: 'SQL (Intermediate) Certificate',
        issuedDate: 'Nov 2021',
        img: 'sqli',
        link: 'https://www.hackerrank.com/certificates/80f91f19cb53',
      },
    ],
  },
  {
    issuing: 'Secure Code Warrior',
    cerList: [
      {
        name: '2021 AppSec Assessment: Injection Flaws - Web',
        issuedDate: 'Sep 2021',
        img: 'secure',
        link: 'https://drive.google.com/file/d/18c8Y6H8jMJ5e9f1j82JMWlG_k-DD1qKz/view?usp=sharing',
      },
    ],
  },
]
