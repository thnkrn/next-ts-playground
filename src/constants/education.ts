export const EDUCATION = [
  {
    title: 'Chualalongkorn University | International School of Engineering',
    duration: '(2015 - 2019)',
    location: 'Bangkok, Thailand',
    type: '(Bachelor of Engineering - BE, Computer Engineering)',
    desc: [`Major in Information and Communication Engineering (ICE)`],
  },
  {
    title: `St. Cuthbert's high school`,
    duration: '(2012 - 2013)',
    location: 'Newcastle, United Kingdom',
    type: '(Student Exchange Program)',
    desc: [`Major in Sci-Math`],
  },
  {
    title: `Bangkok Christian College`,
    duration: '(2003 - 2015)',
    location: 'Bangkok, Thailand',
    type: '(Elementary school - High school)',
    desc: [`Major in Sci-Math`],
  },
]
