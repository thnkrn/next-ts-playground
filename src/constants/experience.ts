export const EXPERIENCE = [
  {
    title: 'Assistant Manager, Full-stack Developer',
    duration: '(Jan 2021 - Present)',
    company: 'Prudential Life Assurance Public Company Ltd. (Thailand)',
    type: '(Full-time)',
    desc: [
      `Move back to full-stack developer after focusing on a front-end side for 2 years, but for the back-end side, I'm working for only the BFF or backend for frontend part, not the mainstream for back-end developer, we having an actual backend team which is served the data for us like AS-400 or Java developer team.`,
      `A further step from the previous position, I'm having new opportunities to help with the guidance of new tech stack or the new principle, helping the team to build up a standardized or improvement both on tech performance and team well-being.`,
      `As a big company, we mostly handle many projects and need to serve fast and go for both internal and external users that involving me to familiar with many scrapfold and how to build a template to make suitable for each project. As mentioned that leads me to have an experience with many tech stuffs and usually need to resarch an up-to-date tech systems.`,
    ],
  },
  {
    title: 'Senior, Software Engineer',
    duration: '(Jul 2020 – Dec 2020)',
    company: 'Pomelo Fashion',
    type: '(Full-time)',
    desc: [
      `As a Senior, I have to take more responsibility not only the coding skill but also for helping the team to manage the tasks and estimate the incoming requirements. From that point, I have a new potential opportunity to offer my opinion in the meeting or have a right to decide something on the project with the other teams.`,
      `Moreover, not focusing on coding the new features and serve users as requirements, I also have to involve in web monitoring, infrastructure monitoring, and manage the outsourcing for web team.`,
    ],
  },
  {
    title: 'Software Engineer',
    duration: '(Jun 2019 – Jun 2020)',
    company: 'Pomelo Fashion',
    type: '(Full-time)',
    desc: [
      `Being a React developer with Next.JS framework and perform on many tasks of requirement. Starting with the marketing part, developed on a landing page or special campaign for Pomelo fashion mostly use a coding skill on the UI part and animation.`,
      `Second, on the product tribe, creating and developing the new features or focusing on develope the product for the business part for this kind of development will mainly focus on the logic part.`,
      ` Following by coding on performance squad, mainly focus on how to develop and make the website be faster and more stable like responsive component, refactor code, redesign component, config file, or every kind of the web performance things.`,
      `Lastly, being a frontend DevOps, managing AWS like Elastic Beanstalk, EC2 or Cloudfront. Creating the CSP for the website and Moving web project from BitBucket to GitHub. Managing and moving web project to the Docker and Kubernetes.`,
    ],
  },
  {
    title: 'Full-stack Developer',
    duration: '(Jun 2018 – Sep 2018)',
    company: 'Prudential Life Assurance Public Company Ltd. (Thailand)',
    type: '(Internship)',
    desc: [
      `Internship for 2 months, being a software and artificial intelligence engineer`,
      `Creating a KYC mobile application that is used for cleansing and managing the customer database. I created a KYC app on Ionic 3 which is the mobile app framework is built on Angular and used JS, SCSS, and HTML to create the application. KYC app consists of 3 functions that use AI for 2 main functions.`,
      `First, I used the Amazon Web Service for Rekognition in order to compare selfie face with the picture in ID card of the customer, the app will return the similarity score for above 90 percent, below that its mean not match. I used Node.JS to create this back-end function and warp with the Express.JS after that send it to the cloud function for using in the application via Heroku.`,
      `Next function, the app will collect the information of customer from ID card by using the OCR function in order to reduce the processing time for users. I used the Google Vision to set the size of the ID card and the boxes of each field to collect all of the user's information.`,
      `Followed by the last function, the app will send the user's information to the Revenue Department of Thailand to check the validity of ID card. The result will return in the status of death, loss, or valid/invalid of the card. This application help the company a lot for cleansing the invalid data because the company had a million user data.`,
    ],
  },
  {
    title: 'Full-stack Developer',
    duration: '(Jun 2017 – Sep 2018)',
    company: 'Fastwork Technologies Co., Ltd.',
    type: '(Freelance)',
    desc: [
      `Working as a freelance developer via studying at the university for developing mobile or web application, mostly focusing on Front-end stuff`,
    ],
  },
]
