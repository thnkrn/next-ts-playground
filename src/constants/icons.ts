const prefix = '/icons/icons-sprite.svg#'

export const ICONS = {
  location: `${prefix}location`,
  github: `${prefix}github`,
  linkedin: `${prefix}linkedin`,
  email: `${prefix}email`,
  work: `${prefix}work`,
}
