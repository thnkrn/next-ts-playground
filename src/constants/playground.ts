export const PLAYGROUND_LIST = [
  {
    title: 'Typography',
    subtitle: 'Simple ways to use a font styling in your website by Typography design system',
    imageName: 'typography',
    linkURL: '/Typography',
  },
  {
    title: 'Image component for static site ',
    subtitle: 'Next/Image not supported static site, nvm!, I have alternative for you',
    imageName: 'image-static-site',
    linkURL: '/Image-static-site',
  },
]
