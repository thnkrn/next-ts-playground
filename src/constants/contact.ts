import { ICONS } from 'constants/icons'

export const CONTACT_LIST = [
  {
    icon: ICONS.email,
    title: 't.ariyagusolsuthi@gmail.com',
  },
  {
    icon: ICONS.github,
    title: 'Github',
    isLink: true,
    link: 'https://github.com/thnkrn',
  },
  {
    icon: ICONS.linkedin,
    title: 'Linkedin',
    isLink: true,
    link: 'https://www.linkedin.com/in/thanakorn-ariyagusolsuthi',
  },
]
