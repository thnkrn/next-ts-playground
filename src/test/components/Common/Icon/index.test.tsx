import Icon from 'components/Common/Icon'
import { ICONS } from 'constants/icons'
import { render } from 'test/util'

describe('Icon', () => {
  describe('Rendering', () => {
    it('should match snapshot', () => {
      const { asFragment } = render(<Icon src={ICONS.email} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
