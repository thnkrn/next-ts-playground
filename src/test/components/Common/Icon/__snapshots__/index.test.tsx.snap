// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Rendering should match snapshot 1`] = `
<DocumentFragment>
  <style
    data-cmpld="true"
  >
    ._1bsb1tcg{width:24px}._4t3i1tcg{height:24px}._tzy4idpf{opacity:0}._18t5kb7n.loaded{opacity:1}
  </style>
  <svg
    class="_1bsb1tcg _4t3i1tcg _tzy4idpf _18t5kb7n icon loaded"
  >
    <use
      href="/icons/icons-sprite.svg#email"
    />
  </svg>
</DocumentFragment>
`;
